File: Week2/hyochang/B3273.java
package hyochang;
import java.util.Arrays;
import java.util.Scanner;
public class B3273 {
        public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt(); 
        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        Arrays.sort(arr); 

        int x = sc.nextInt();

        int left = 0;
        int right = n - 1;
        int cnt = 0; //해당 쌍의 갯수  

        while (left < right) {//최소가 최대를 넘길 때까지
            int sum = arr[left] + arr[right];

            if (sum == x) {//같으면 범위를 줄여서 다시 탐색
                cnt++;
                left++; 
                right--;
            } else if (sum < x) {//결과와 목표에 따른 탐색 반경 조정
                left++; 
            } else {
                right--;
            }
        }

        System.out.println(cnt);
    }
}

File: Week2/KimSungHa/B3273.py
# 문제: 3273번 두 수의 합
# 설명: 서로 다른 양의 정수들로 이뤄진 수열에서 두 수의 합이 주어진 x와 일치하는 쌍의 갯수 찾기

n = int(input()) # n, arr, x 입력
arr = list(map(int,input().split()))
x = int(input())
cnt = 0 # 만족하는 쌍의 수를 셀 정수 할당

for i in range(n-1):  # 각각의 수를 돌며 (두 쌍을 만들 것이므로 맨 끝은 시작점으로 잡지 않아도 된다.)
    for j in range(i+1,n):  # 이 수 다음 자리부터 끝까지 (이번엔 끝까지)
        if arr[i]+arr[j] == x:  # 두 수의 합이 x와 일치하면 cnt에 1을 더한다.
            cnt += 1

print(cnt)